### PR TITLE
update specs for checking returned components

### PR DESCRIPTION
**Pull request by S-anasol**
Added few tests.

Maybe we should check more? I have no idea about other checks :)

> SUCCESS: 15 specs, 0 failures, 0 skipped, 0 disabled in 11.034s.

### CODE REVIEW

File: specs/specs.js [hunk 1]
```diff
@@ -89,5 +89,32 @@ describe("Fingerprint2", function () {
         });
       });
     });
+	
+	describe("return components", function () {
```
Comment (Valve) [inline specs/specs.js]: please fix indentation

File: specs/specs.js [hunk 2]
```diff
@@ -89,5 +89,32 @@ describe("Fingerprint2", function () {
...
+      it("does it return components", function (done) {
```
Comment (Valve) [inline specs/specs.js]: should be imperative, e.g. it("returns components as a second argument to callback")

File: specs/specs.js [hunk 3]
```diff
@@ -89,5 +89,32 @@ describe("Fingerprint2", function () {
...
+        var fp2 = new Fingerprint2();
+        fp2.get(function(result, components) {
+          expect(components).not.toBeNull();
+          done();
+        });
+      });
+	  
+      it("is first element user_agent", function (done) {
```
Comment (Valve) [inline specs/specs.js]: it ("returns user_agent as the first element"). i.e. the spec should read as a sentence

File: specs/specs.js [hunk 4]
```diff
@@ -89,5 +89,32 @@ describe("Fingerprint2", function () {
...
+        var fp2 = new Fingerprint2();
+        fp2.get(function(result, components) {
+          expect(components[0].key).toEqual("user_agent");
+          done();
+        });
+      });
+	  
+      it("user_agent is PhantomJS", function (done) {
```
Comment (Valve) [inline specs/specs.js]: I would advise agains hardcoding PhantomJS, because we may not use it like tomorrow

**Comment by Valve** [comment]
Looks good. Please add a check that the components is an array and that some of its individual elements are arrays too (fonts, plugins etc).

**Comment by S-anasol** [comment]
How can i check that components is an array? :smile: 
Found only toBeArray in jasmine-expect, but have no idea how include it to tests.

Never used JS tests before :)

File: specs/specs.js (hunk 4 above)
Comment (S-anasol) [inline specs/specs.js]: okay, removed browser test

File: specs/specs.js (hunk 3 above)
Comment (S-anasol) [inline specs/specs.js]: done

File: specs/specs.js (hunk 2 above)
Comment (S-anasol) [inline specs/specs.js]: done

File: specs/specs.js (hunk 1 above)
Comment (S-anasol) [inline specs/specs.js]: done

**Comment by S-anasol** [comment]
added jasmine-matchers for testing array/arrayOfObjects

File: specs/specs.js [hunk 5]
```diff
@@ -89,5 +89,45 @@ describe("Fingerprint2", function () {
...
+          done();
+        });
+      });
+      
+      it("check if js_fonts component is array", function (done) {
+        var fp2 = new Fingerprint2();
+        fp2.get(function(result, components) {
+          for(var x = 0; x < components.length; x++) {
+            if(components[x].key == "js_fonts")
+            {
```
Comment (Valve) [inline specs/specs.js]: open curly on the same line please

File: specs/specs.js [hunk 6]
```diff
@@ -89,5 +89,45 @@ describe("Fingerprint2", function () {
         });
       });
     });
+    
+    describe("returns components as a second argument to callback", function () {
+      it("does it return components", function (done) {
```
Comment (Valve) [inline specs/specs.js]: it("returns components")

File: specs/specs.js [hunk 7]
```diff
@@ -89,5 +89,45 @@ describe("Fingerprint2", function () {
...
+        var fp2 = new Fingerprint2();
+        fp2.get(function(result, components) {
+          expect(components).not.toBeNull();
+          done();
+        });
+      });
+      
+      it("check if returned components is array", function (done) {
```
Comment (Valve) [inline specs/specs.js]: it("checks if..

File: specs/specs.js [hunk 8]
```diff
@@ -89,5 +89,45 @@ describe("Fingerprint2", function () {
...
+        var fp2 = new Fingerprint2();
+        fp2.get(function(result, components) {
+          expect(components).toBeArrayOfObjects();
+          done();
+        });
+      });
+      
+      it("check if js_fonts component is array", function (done) {
```
Comment (Valve) [inline specs/specs.js]: check => checks

**Comment by S-anasol** [comment]
done :+1:

**Comment by Valve** [comment]
thanks

**Pull request merged**